js/client-teambuilder.js

```diff
@@ -1699,19 +1699,25 @@
 
 			if (template.speciesid === 'meowstic') {
 				smogdexid = 'meowstic-m';
-			} else if (smogdexid === 'rotom' || smogdexid === 'deoxys' || smogdexid === 'kyurem' || smogdexid === 'giratina' || smogdexid === 'shaymin' || smogdexid === 'tornadus' || smogdexid === 'thundurus' || smogdexid === 'landorus' || smogdexid === 'pumpkaboo' || smogdexid === 'gourgeist' || smogdexid === 'arceus' || smogdexid === 'meowstic' || smogdexid === 'hoopa') {
-				if (template.forme) smogdexid += '-' + toId(template.forme);
+			} else if (template.forme) {
+				if (template.speciesid === 'vivillon-fancy' || template.speciesid === 'vivillon-pokeball') smogdexid = 'vivillon';
```

Comment:
.... @Zarel is right... IDs only have hyphens if they are room ids

Just press F12:

``> Tools.getTemplate('vivillon-fancy').speciesid``
``< "vivillonfancy"``